airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/upgrade_java_cdk/pipeline.py

```diff
@@ -0,0 +1,99 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+from __future__ import annotations
+
+import os
+import re
+from typing import TYPE_CHECKING
+
+from pipelines.airbyte_ci.connectors.context import ConnectorContext
+from pipelines.airbyte_ci.connectors.reports import ConnectorReport, Report
+from pipelines.helpers import git
+from pipelines.models.steps import Step, StepResult, StepStatus
+
+if TYPE_CHECKING:
+    from anyio import Semaphore
+
+
+class SetJavaCDKVersion(Step):
```

Comment:
This class should be unit tested,[ feel free to take inspiration from what we did for format](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/tests/test_format/test_commands.py#L1), as it's also mutating local files.